### PR TITLE
Add LogFailedExpectation

### LINKED ISSUE #57: Support log.Diff to represent expectation mismatch
Expected API:

`logger.Error("protocol version mismatch", log.Diff(protocol, expectedProtocol)`

### LINKED ISSUE #121: ResultsBlockHeader in CommitTransactionReceipts is too wide in scope
`transactionpool.CommitTransactionReceipts` provides an instance of `ResultsBlockHeader` - out of which transaction pool only needs last block height and timestamp.

This creates unnecessary coupling and provides too wide a scope for transaction pool.

Please introduce changes in proto to *only* pass the block height and timestamp.

### PR DESCRIPTION
**Pull request by yonidavidson**
This function allows you to log a diff between 2 fields.
Can provide the base to fix #57

### CODE REVIEW

File: instrumentation/log/basic_logger.go [hunk 1]
```diff
@@ -264,6 +265,13 @@ func (b *basicLogger) Error(message string, params ...*Field) {
 	b.Log("error", message, params...)
 }
 
+func (b *basicLogger) LogFailedExpectation(message string, expected *Field, actual *Field, params ...*Field) {
+	actual.Key = "actual-" + actual.Key
+	expected.Key = "expected-" + expected.Key
+	newParams:= append(params,expected,actual)
```
Comment (jlevison) [inline instrumentation/log/basic_logger.go]: we should also support the case where params is nil (so only the expected/actual is provided)

Comment (netoneko) [reply instrumentation/log/basic_logger.go]: It will work because `params` are always present even if they are not populated with anything.

**Comment by electricmonk** [comment]
What’s the use case for this? Meaning - what problem are we solving with logging diffs? Or failed expectations?

**Comment by netoneko** [comment]
@electricmonk we have many checks and validations in multiple places and we would like to have universal way to debug if they fail. This pr provides exactly that.

**Pull request merged**